### PR TITLE
[WIP] Updating for release 2.1

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

File: nservicebus/sqlserver/concurrency.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Prior to 2.0 there was no support for callbacks in the SQLServer transport. There is a single queue table for each logical endpoint, no matter how many actual endpoint instances were running. Each endpoint running SQLServer transport spins up a fixed number of threads (controlled by `MaximumConcurrencyLevel` property of `TransportConfig` section). Each thread runs a loop, polling the database for messages awaiting processing.
+
+The disadvantage of this simple model is the fact that satellites (e.g. Second-Level Retries, Timeout Manager) share the same concurrency settings but usually have much lower throughput requirements. If both SLR and TM are enabled, setting `MaximumConcurrencyLevel` to 10 results in 40 threads in total, each polling the database even if there are no messages to be processed.
+
+### 2.0
+
+In released 2.0 we added support for callbacks. Callbacks are implemented by each endpoint instance having a unique[secondary queue](http://docs.particular.net/nservicebus/sqlserver/configuration#secondary-queues). The receive for the secondary queue does not use the `MaximumConcurrencyLevel` and defaults to 1 thread. This value can be adjusted via the config API.
```
Comment (SimonCropp) [inline nservicebus/sqlserver/concurrency.md]: links should be relative

File: nservicebus/sqlserver/multiple-databases.md [hunk 2]
```diff
@@ -1,14 +1,83 @@
...
+### Other endpoints
+
+If a particular remote endpoint requires customization of any part of the routing (schema, database or instance name/address), appropriate values have to be provide either via code or via configuration convention.
+
+#### Push mode
+
+In the push mode the whole collection of endpoint connection information objects is passed during configuration time.
+
+```
+busConfig.UseTransport<SqlServerTransport>().UseSpecificConnectionInformation(
```
Comment (SimonCropp) [inline nservicebus/sqlserver/multiple-databases.md]: can this be moved to a snippet

File: nservicebus/sqlserver/multiple-databases.md [hunk 3]
```diff
@@ -1,14 +1,83 @@
...
+      .UseConnectionString("SomeConnectionString")
+))
+```
+
+#### Pull mode
+
+The pull mode can be used when specific information is not available at configuration time. He can pass a `Func<String, ConnectionInfo>` that will be used by the SQLServer transport to resolve connection information at runtime.
+
+```
+busConfig.UseTransport<SqlServerTransport>()
```
Comment (SimonCropp) [inline nservicebus/sqlserver/multiple-databases.md]: can this be moved to a snippet

File: nservicebus/sqlserver/multiple-databases.md (hunk 2 above)
Comment (SzymonPobiega) [inline nservicebus/sqlserver/multiple-databases.md]: Yeah. I put it here because it is not yet released.

**Pull request merged**